node/src/components/linear_chain_sync.rs

```diff
@@ -449,14 +506,34 @@ where
                         }
                     }
                     BlockByHeightResult::FromStorage(block) => {
-                        // We shouldn't get invalid data from the storage.
-                        // If we do, it's a bug.
-                        assert_eq!(block.height(), block_height, "Block height mismatch.");
-                        trace!(%block_height, "Linear block found in the local storage.");
-                        // When syncing descendants of a trusted hash, we might have some of them in
-                        // our local storage. If that's the case, just
-                        // continue.
-                        self.block_downloaded(rng, effect_builder, &block)
+                        match self.state {
+                            State::SyncingTrustedHash {
+                                highest_block_header: Some(ref hdr),
+                                ref validator_weights,
+                                ..
+                            } => {
+                                info!(%block_height, "Read a block from storage while syncing up\
+                                    to trusted hash; Starting at the highest known block\
```

Comment:
All whitespace in the next line after the backslash is ignored. You need a space before it, otherwise it will be `upto` and `blockinstead`.
```suggestion
                                info!(%block_height, "Read a block from storage while syncing up \
                                    to trusted hash; Starting at the highest known block \
```